collector/cpu_linux.go

```diff
@@ -77,7 +78,7 @@ func NewCPUCollector() (Collector, error) {
 
 // Update implements Collector and exposes cpu related metrics from /proc/stat and /sys/.../cpu/.
 func (c *cpuCollector) Update(ch chan<- prometheus.Metric) error {
-	if *enableCPUInfo {
+	if *enableCPUInfo && (runtime.GOARCH == "amd64") {
```

Comment:
can we do `runtime.GOARCH != "s309x"` instead?

Reply:
This is a problem for PPC too. And  i am not sure if ARM or other arches will have the same problem...so i just enabled it only for x86

Reply:
the parsecpuinfo function is x86 specific which is also why I excluded for other arches

Reply:
From what I can tell on PPC and other platforms it shouldn't crash, but it won't have all the same info available.  I think the suggested change to only activate on amd64 makes sense for now.  We can add back other platforms in the future as needed with better testing (possibly upstream).

Reply:
PPC doesn't crash, but it doesn't parse correctly either. s390x crashes because the format is different..